### PR TITLE
Apply provider default labels to handwritten resources

### PR DESCRIPTION
**Pull request by zli82016**
With the new model, the test for the following two resources failed. I am going to migrate these two resources later.
`google_compute_instance_template` has `forcenew: true`. 
`google_compute_region_instance_template` has `Set:  schema.HashString`

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/services/bigquery/resource_bigquery_table_test.go [hunk 1]
```diff
@@ -3098,7 +3098,7 @@ resource "google_bigquery_table" "test" {
 
   friendly_name = "bigquerytest"
   labels = {
-    "terrafrom_managed" = "true"
+    "terrafrom_managed" = "false"
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/bigquery/resource_bigquery_table_test.go:3101]: Are we changing this to test update?

File: mmv1/third_party/terraform/tpgresource/labels.go [hunk 2]
```diff
@@ -8,10 +8,10 @@ import (
 	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
 )
 
-func FlattenLabels(labels map[string]string, d *schema.ResourceData) map[string]interface{} {
+func FlattenLabels(labels map[string]string, d *schema.ResourceData, linage string) map[string]interface{} {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tpgresource/labels.go:11]:
```suggestion
func FlattenLabels(labels map[string]string, d *schema.ResourceData, lineage string) map[string]interface{} {
```

File: mmv1/third_party/terraform/tpgresource/labels.go [hunk 3]
```diff
@@ -8,10 +8,10 @@ import (
...
 	transformed := make(map[string]interface{})
 
-	if v, ok := d.GetOk("labels"); ok {
+	if v, ok := d.GetOk(linage); ok {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tpgresource/labels.go:14]:
```suggestion
	if v, ok := d.GetOk(lineage); ok {
```

File: mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go [hunk 4]
```diff
@@ -1405,9 +1412,12 @@ func resourceBigQueryTableRead(d *schema.ResourceData, meta interface{}) error {
 	if err := d.Set("max_staleness", res.MaxStaleness); err != nil {
 		return fmt.Errorf("Error setting max_staleness: %s", err)
 	}
-	if err := d.Set("labels", tpgresource.FlattenLabels(res.Labels, d)); err != nil {
+	if err := d.Set("labels", tpgresource.FlattenLabels(res.Labels, d, "labels")); err != nil {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go:1415]: We may want to wrap the whole `d.Set` call into a function in order to assert that the target field and the `lineage` values match. It's a pretty easy copy-paste error to let those diverge at the moment.

**Review by rileykarson — APPROVED** [review]

File: mmv1/third_party/terraform/services/bigquery/resource_bigquery_table_test.go (hunk 1 above)
Comment (zli82016) [reply mmv1/third_party/terraform/services/bigquery/resource_bigquery_table_test.go:3101]: Yeah, that is the reason.

File: mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go (hunk 4 above)
Comment (zli82016) [reply mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go]: Done. A new func SetLabels is called here.

File: mmv1/third_party/terraform/tpgresource/labels.go [hunk 5]
```diff
@@ -8,18 +8,18 @@ import (
 	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
 )
 
-func FlattenLabels(labels map[string]string, d *schema.ResourceData) map[string]interface{} {
+func SetLabels(labels map[string]string, d *schema.ResourceData, lineage string) error {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tpgresource/labels.go:11]: nit: Do you mind commenting the function, to explain the behaviour?

Comment (zli82016) [reply mmv1/third_party/terraform/tpgresource/labels.go:16]: Sure. Added doc.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**